### PR TITLE
imxrt-flash: non-XIP flash ops don't block XIP threads

### PR DESCRIPTION
**Pull request by ziemleszcz**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Chore (refactoring, style fixes, git/CI config, submodule management, no code logic changes)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (imxrt1176).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Review by agkaminski — APPROVED** [review]
LGTM

File: storage/imxrt-flash/fspi.c [hunk 1]
```diff
@@ -109,41 +76,104 @@ static int flexspi_checkFlags(flexspi_t *fspi)
...
+
+	return EOK;
+}
+
+
+void flexspi_swreset(flexspi_t *fspi)
+{
+	*(fspi->base + mcr0) |= 1u;
+	flexspi_poll(fspi, fspi->base + mcr0, 1u, 0u, 0, 0, 0);
+}
```
Comment (gerard5) [inline storage/imxrt-flash/fspi.c:115]: Not used, remove dead code.

File: storage/imxrt-flash/fspi.c [hunk 2]
```diff
@@ -109,41 +76,104 @@ static int flexspi_checkFlags(flexspi_t *fspi)
...
-	uint8_t *end = ptr + xfer->data.read.sz;
+	int retries = 0;
 
-	while (ptr != end) {
-		volatile uint8_t *rfdr = (volatile uint8_t *)(fspi->base + rfdr32); /* 2x */
-
-		/* Wait for rx FIFO available */
-		while ((*(fspi->base + intr) & (1u << 5u)) == 0u) {
+	while ((*addr & mask) != val) {
+		if (checkFlags) {
```
Comment (gerard5) [inline storage/imxrt-flash/fspi.c:85]: Use `bool checkFlags`, or:
```suggestion
		if (checkFlags != 0) {
```

File: storage/imxrt-flash/nor.c [hunk 3]
```diff
@@ -103,12 +103,16 @@ int nor_waitBusy(flexspi_t *fspi, uint8_t port, time_t timeout)
...
+	for (;;) {
 		res = nor_readStatus(fspi, port, &status, timeout);
 		if (res < EOK) {
 			return res;
 		}
-	} while ((status & 1) != 0);
+		if ((status & 1) == 0) {
+			break;
+		}
+		flexspi_schedYield(fspi);
```
[the remark this answers is not in the document]
Comment (ziemleszcz) [reply storage/imxrt-flash/nor.c:114]: Not exactly true, please notice introduction of quick poll.

File: storage/imxrt-flash/fspi.c [hunk 4]
```diff
@@ -29,6 +29,9 @@
  */
 
 
+#define QUICK_POLL_MAX_RETRIES 5
```
Comment (gerard5) [inline storage/imxrt-flash/fspi.c:32]:
```suggestion
#ifndef QUICK_POLL_MAX_RETRIES
#define QUICK_POLL_MAX_RETRIES 5
#endif
```
Check if accessible from `boar_config.h`

**Review by agkaminski — APPROVED** [review]

**Review by gerard5 — APPROVED** [review]

**Pull request merged**